sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/EncryptOptions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    /// <summary>
+    /// Options for encrypting with a symmetric key including <see cref="EncryptionAlgorithm.A128Gcm"/>, <see cref="EncryptionAlgorithm.A192Gcm"/>, or <see cref="EncryptionAlgorithm.A256Gcm"/>.
+    /// </summary>
+    public class EncryptOptions : IJsonSerializable
+    {
+        /// <summary>
+        /// Creates an instance of the <see cref="EncryptOptions"/> class for use with symmetric keys including <see cref="EncryptionAlgorithm.A128Gcm"/>, <see cref="EncryptionAlgorithm.A192Gcm"/>, or <see cref="EncryptionAlgorithm.A256Gcm"/>.
+        /// </summary>
+        /// <param name="plaintext">The required plaintext to encrypt.</param>
+        /// <param name="iv">The initialization vector for encryption. If null, one will be generated like an nonce and returned in <see cref="EncryptResult.Iv"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="plaintext"/> is null.</exception>
+        public EncryptOptions(byte[] plaintext, byte[] iv = null)
```

Comment:
@KrzysztofCwalina @tg-msft @schaabs Scott and I talked about this and originally we thought maybe an algorithm-specific class like `AesEncryptionOptions` (and make this class abstract) might help delineat options more, but as I was working on that it just wasn't very discoverable. By defining a constructor like this - including the optional param - I think we can better inform the user what to pass. If we should add other algorithm-specific variants in the future, we can add constructors.

I think this is better than, say, having an overload for each algorithm difference. In time, that will lead to many different overloads, though admittedly they should be fairly obvious which to use. Ultimately, too, internally we'll converge on a single request object but I know that's of little importance (i.e. just pointing out they all funnel to a single REST API anyway).

Thougths?